tools/c7n_mailer/c7n_mailer/email_delivery.py

```diff
@@ -28,11 +30,24 @@ class EmailDelivery(object):
     def __init__(self, config, session, logger):
         self.config      = config
         self.logger      = logger
+        self.session     = session
         self.aws_ses     = session.client('ses', region_name=config.get('ses_region'))
         self.ldap_lookup = self.get_ldap_connection()
 
     def get_ldap_connection(self):
         if self.config.get('ldap_uri'):
+            try:
+                if self.config.get('ldap_bind_password', None):
+                    kms = self.session.client('kms')
+                    self.config['ldap_bind_password'] = kms.decrypt(
+                        CiphertextBlob=base64.b64decode(self.config['ldap_bind_password']))[
```

Comment:
so we might need one more exception catch here, for a non kms password, thats not base64 encoded, if base64 decode will throw an error, that would be pre the decrypt api call. worth trying out and verifying before merging.

```
>>> import base64
>>> base64.b64decode('??seaf231230923q8>>:')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/base64.py", line 76, in b64decode
    raise TypeError(msg)
TypeError: Incorrect padding
>>> 
```

Reply:
Ah yes, and it looks like we need to also catch `binascii.Error` for python 3.